src/Management/src/Endpoint/Actuators/Health/HealthEndpointMiddleware.cs

```diff
@@ -52,20 +52,19 @@ private string GetRequestedHealthGroup(PathString requestPath)
     {
         string[] requestComponents = requestPath.Value?.Split('/') ?? [];
 
-        if (requestComponents.Length > 0)
+        if (requestComponents.Length > 0 && requestComponents[^1] != _endpointOptionsMonitor.CurrentValue.Id)
```

Comment:
Please take a single snapshot of the options monitor per request and pass it downstream to get a coherent result.